### PR TITLE
Add scraping layer

### PR DESCRIPTION
**Pull request by soulages04**

### CODE REVIEW

**Comment by soulages04** [comment]
Hola gregory!
Here is an updated version with gitignore

File: scripts/17ft.js [hunk 1]
```diff
@@ -2,54 +2,52 @@ const request = require("request");
...
+
+  // push array into forecast object
+  let forecast = {
+    bouy : report[0],
+    swell_height : report[1],
+    period : report[2],
+    swell_direction : report[3],
+    swell_heiht : report[4],
+    wind_speed : report[5],
+    water_temp : report[6],
```
Comment (gsoulages-rms) [inline scripts/17ft.js]: shouldn't have trailing `,`

File: scripts/17ft.js [hunk 2]
```diff
@@ -2,54 +2,52 @@ const request = require("request");
...
+    }
+  });
+
+  // push array into forecast object
+  let forecast = {
+    bouy : report[0],
+    swell_height : report[1],
+    period : report[2],
+    swell_direction : report[3],
+    swell_heiht : report[4],
```
Comment (gsoulages-rms) [inline scripts/17ft.js]: misspelled height

File: scripts/17ft.js [hunk 3]
```diff
@@ -2,54 +2,52 @@ const request = require("request");
...
+        report.push($(this).text());
+        }
+      });
+    }
+  });
+
+  // push array into forecast object
+  let forecast = {
+    bouy : report[0],
+    swell_height : report[1],
```
Comment (gsoulages-rms) [inline scripts/17ft.js]: isn't javascript usually in camelCase or is underscore as separator common?

File: scripts/17ft.js [hunk 4]
```diff
@@ -2,54 +2,52 @@ const request = require("request");
...
+  request.get(url, (error, response, body) => {
+  let report = [];
+
+  $ = cheerio.load(body)
+
+  $('td').each(function(i, elem) {
+    // Filter table by name, skip first sibling (its useless)
+    // remove extraneous characters, and push into array
+    if ($(this).text() === name) {
+      let siblings = $(this).next().siblings();
```
Comment (gsoulages-rms) [inline scripts/17ft.js]: consider using `.map( function(index, element) )` rather than `each`

Comment (gsoulages-rms) [reply scripts/17ft.js]: this will avoid the need for the report array

File: scripts/17ft.js [hunk 5]
```diff
@@ -2,54 +2,52 @@ const request = require("request");
...
+      siblings.each(function (i, elem) {
+        if (i > 0) {
+          let num = $(this).text().replace(/[^0-9-.]/g, "");
+          report.push(num);
+        } else {
+        report.push($(this).text());
```
Comment (gsoulages-rms) [inline scripts/17ft.js]: isn't indented correctly (and you have the buoy name already)

File: scripts/17ft.js [hunk 6]
```diff
@@ -2,54 +2,52 @@ const request = require("request");
...
-// westSb = array.eq(48).children()
-// wind =
-// speed: westSb.eq(5).text()
-// direction: westSb.eq(6).text()
-
-// {swell, wind}
\ No newline at end of file
+function getBouy(name) {
+  
+  request.get(url, (error, response, body) => {
```
[the remark this answers is not in the document]
Comment (soulages04) [reply scripts/17ft.js]: hey bro i put that up in the top of the file, no ?

File: scripts/17ft.js [hunk 7]
```diff
@@ -2,54 +2,52 @@ const request = require("request");
...
+  let report = [];
```
[the remark this answers is not in the document]
Comment (soulages04) [reply scripts/17ft.js]: updated to indent function body

File: scripts/17ft.js [hunk 8]
```diff
@@ -2,54 +2,52 @@ const request = require("request");
...
+
+  $ = cheerio.load(body)
+
+  $('td').each(function(i, elem) {
```
[the remark this answers is not in the document]
Comment (soulages04) [reply scripts/17ft.js]: looks like cheerio doesn't like es6! 
https://github.com/cheeriojs/cheerio/issues/762

i tried using the alternative 'element' parameter but it wasn't working. waste of time for now to try to fix

[the remark this answers is not in the document]
Comment (soulages04) [reply scripts/17ft.js]: im going to compile all refactor changes into another chunk of work, which ill get back to after completing the caching layer. implementing filter will be part of this refactoring effort.

File: scripts/17ft.js (hunk 4 above)
Comment (soulages04) [reply scripts/17ft.js]: adding to my refactoring chunk

File: scripts/17ft.js (hunk 5 above)
Comment (soulages04) [reply scripts/17ft.js]: fixed indent (and no i dont because i skip the first sibling)

File: scripts/17ft.js (hunk 3 above)
Comment (soulages04) [reply scripts/17ft.js]: yes, i prefer underscore for json objects but *i changed it*

File: scripts/17ft.js (hunk 2 above)
Comment (soulages04) [reply scripts/17ft.js]: fixed

File: scripts/17ft.js (hunk 1 above)
Comment (soulages04) [reply scripts/17ft.js]: fixed.

**Comment by soulages04** [comment]
thanks for these comments bro! 
i've added a list item in my * TODO * for my refactor changes. will get back to those after completing after the caching layer.

merging this to develop

**Pull request merged**